# Information about maize inbred genotypes

I found two important references:

* [Romay et al. 2013](https://link.springer.com/article/10.1186/gb-2013-14-6-r55)
* [The original Goodman study](https://onlinelibrary.wiley.com/doi/full/10.1111/j.1365-313X.2005.02591.x)

## Romay paper (2013)

Importing the Romay supplementary table:

In [2]:
import pandas as pd

#genotype_info_romay_df = pd.read_csv("/home/rsantos/Repositories/maize_microbiome_transcriptomics/genotype_information/gb-2013-14-6-r55-S1.csv", sep='\t')
genotype_info_romay_df = pd.read_csv("/home/santosrac/Repositories/maize_microbiome_transcriptomics/genotype_information/gb-2013-14-6-r55-S1.csv", sep='\t')

genotype_info_romay_df.set_index('Inbred line', inplace=True)
genotype_info_romay_df.head()

Accesion N  N GBS samples  N Plants  Avg. IBS  % missing  \
Inbred line                                                            
42            PI233312              1         1       NaN       0.47   
66           Ames26770              1         1       NaN       0.56   
90            NSL30903              1         1       NaN       0.52   
176           PI233313              1         1       NaN       0.67   
207           PI601005              2         2     0.996       0.50   

            Breeding program    Pop structure  
Inbred line                                    
42                     Other       sweet corn  
66                     Other     unclassified  
90                     Other     unclassified  
176                    Other       sweet corn  
207                    ExPVP  non-stiff stalk

In [3]:
#genotype_info_wallace2018_df = pd.read_csv("/home/rsantos/Repositories/maize_microbiome_transcriptomics/correlations_rnaseq_metataxonomics/0_kremling_genotype_plot_day_key.txt", sep='\t')
genotype_info_wallace2018_df = pd.read_csv("/home/santosrac/Repositories/maize_microbiome_transcriptomics/correlations_rnaseq_metataxonomics/0_kremling_genotype_plot_day_key.txt", sep='\t')

genotype_info_wallace2018_df.set_index('Genotype', inplace=True)
genotype_info_wallace2018_df.head()


Plot_Day
Genotype             
33-16     X14A0309_26
38-11      X14A0233_8
4226      X14A0079_26
4722       X14A0311_8
A188       X14A0021_8

In [4]:
merged_df = genotype_info_wallace2018_df.merge(genotype_info_romay_df, left_index=True, right_index=True, how='left')

In [5]:
merged_df.tail(n=30)

Plot_Day Accesion N  N GBS samples  N Plants  Avg. IBS  \
Genotype                                                               
SG18        X14A0321_8   NSL75979            3.0       2.0     0.995   
T232       X14A0285_26   PI550522            5.0       4.0     0.998   
T234       X14A0349_26  Ames27190            3.0       2.0     0.998   
T8          X14A0255_8   PI550518            5.0       4.0     0.998   
Tx303      X14A0443_26  Ames19327           11.0       5.0     0.995   
Tzi11      X14A0491_26   PI540745            3.0       2.0     0.993   
Tzi16      X14A0423_26   PI540747            4.0       3.0     0.994   
Tzi18      X14A0477_26   PI506253            6.0       5.0     0.976   
Tzi25      X14A0419_26   PI506255            4.0       3.0     0.976   
Tzi9       X14A0463_26   PI506247            4.0       3.0     0.985   
U267Y      X14A0331_26  Ames27191            3.0       2.0     0.997   
Va102       X14A0119_8      Va102            2.0       1.0     0.999   
Va14        X14A0395_8  Ames27192            3.0       2.0     0.996   
Va17        X14A0257_8  Ames19002            4.0       3.0     0.998   
Va22        X14A0247_8  Ames19328            4.0       3.0     0.997   
Va26        X14A0109_8   PI587149            3.0       2.0     0.998   
Va35        X14A0159_8   PI587150            5.0       4.0     0.998   
Va59        X14A0105_8  Ames19016            5.0       4.0     0.997   
Va85       X14A0391_26  Ames27193            3.0       2.0     0.995   
Va99        X14A0191_8  Ames24989            3.0       2.0     0.998   
VaW6        X14A0197_8  Ames19000            6.0       5.0     0.993   
W117HT      X14A0043_8   Ames8399            3.0       2.0     0.998   
W153R       X14A0069_8  PI587154             5.0       4.0     0.996   
W182B       X14A0017_8  Ames19330            4.0       3.0     0.993   
W22R-rstd  X14A0240_26   NSL30067            3.0       2.0     0.977   
W22R-rstd   X14A0240_8   NSL30067            3.0       2.0     0.977   
W22         X14A0181_8   NSL30053           11.0       5.0     0.993   
W64A        X14A0121_8   PI587152            4.0       3.0     0.994   
Wf9         X14A0129_8  Ames19293            3.0       2.0     0.998   
Yu796NS     X14A0127_8  Ames27196            3.0       2.0     0.973   

           % missing Breeding program    Pop structure  
Genotype                                                
SG18            0.49            Other          popcorn  
T232            0.25            Other     unclassified  
T234            0.46            Other     unclassified  
T8              0.26            Other     unclassified  
Tx303           0.21            Other     unclassified  
Tzi11           0.35          Nigeria         tropical  
Tzi16           0.30          Nigeria         tropical  
Tzi18           0.28          Nigeria         tropical  
Tzi25           0.30          Nigeria         tropical  
Tzi9            0.33          Nigeria         tropical  
U267Y           0.36            Other     unclassified  
Va102           0.49            Other     unclassified  
Va14            0.47            Other     unclassified  
Va17            0.32            Other     unclassified  
Va22            0.31            Other  non-stiff stalk  
Va26            0.34            Other  non-stiff stalk  
Va35            0.50            Other  non-stiff stalk  
Va59            0.30            Other  non-stiff stalk  
Va85            0.53            Other     unclassified  
Va99            0.42            Other  non-stiff stalk  
VaW6            0.32            Other     unclassified  
W117HT          0.41        Wisconsin     unclassified  
W153R           0.25        Wisconsin  non-stiff stalk  
W182B           0.29        Wisconsin     unclassified  
W22R-rstd       0.36        Wisconsin  non-stiff stalk  
W22R-rstd       0.36        Wisconsin  non-stiff stalk  
W22             0.19        Wisconsin  non-stiff stalk  
W64A            0.29        Wisconsin      stiff stalk  

In [6]:
# Counting the number of genotypes for each group
merged_df['Pop structure'].value_counts()

Pop structure
unclassified       151
stiff stalk         39
tropical            38
non-stiff stalk     27
popcorn              7
sweet corn           5
landraces            2
Name: count, dtype: int64

In [7]:
merged_df['Breeding program'].value_counts()

Breeding program
Other             103
North Carolina     43
Iowa               32
Mexico             28
Minnesota          15
Missouri            7
Wisconsin           7
Indiana             6
Ontario             5
Nigeria             5
Thailand            5
Michigan            3
South Africa        3
Illinois            3
France              1
Spain               1
Nebraska            1
North Dakota        1
Name: count, dtype: int64

In [8]:
merged_df.loc[merged_df['Pop structure'].isna()]

Plot_Day Accesion N  N GBS samples  N Plants  Avg. IBS  \
Genotype                                                              
CML411    X14A0566_26        NaN            NaN       NaN       NaN   
CML504    X14A0569_26        NaN            NaN       NaN       NaN   
CML505     X14A0570_8        NaN            NaN       NaN       NaN   
CML511    X14A0571_26        NaN            NaN       NaN       NaN   
CML84     X14A0572_26        NaN            NaN       NaN       NaN   
CML85     X14A0573_26        NaN            NaN       NaN       NaN   
CML96     X14A0574_26        NaN            NaN       NaN       NaN   

          % missing Breeding program Pop structure  
Genotype                                            
CML411          NaN              NaN           NaN  
CML504          NaN              NaN           NaN  
CML505          NaN              NaN           NaN  
CML511          NaN              NaN           NaN  
CML84           NaN              NaN           NaN  
CML85           NaN              NaN           NaN  
CML96           NaN              NaN           NaN

In [11]:
merged_df['Pop structure'].isna().sum()

7

In [13]:
# After exporting, I (RACS) modified the names to remove X added by R (data.frame column names)
merged_df.to_csv('/home/santosrac/Repositories/maize_microbiome_transcriptomics/genotype_information/wallace_et_al_2018_group_assignments_romay2013.tsv', sep='\t')

## Flint-Garcia paper (2005)


In [27]:
import pandas as pd

genotype_info_flintgarcia_df = pd.read_csv("/home/rsantos/Repositories/maize_microbiome_transcriptomics/genotype_information/flint_garcia_s1_table.csv", sep='\t')

genotype_info_flintgarcia_df.set_index('Inbred', inplace=True)
genotype_info_flintgarcia_df.head()

State/Country                    Pedigree    NSS     SS     TS Pop A  \
Inbred                                                                        
796       Yugoslavia      Vukovarski Yellow Dent  0.785  0.189  0.026     0   
4226        Illinois                 Funk 90 Day  0.917  0.071  0.012     0   
4722         Indiana              SA x Ohio yel.      0      0      0     1   
33-16        Indiana   Lux Johnson Country white  0.972  0.014  0.014     0   
38-11        Indiana  Outcross in line from 176A  0.993  0.003  0.004     0   

       Sweet A Subpopulation  
Inbred                        
796          0         mixed  
4226         0           nss  
4722         0       popcorn  
33-16        0           nss  
38-11        0           nss

In [28]:
genotype_info_wallace2018_df = pd.read_csv("/home/rsantos/Repositories/maize_microbiome_transcriptomics/correlations_rnaseq_metataxonomics/0_kremling_genotype_plot_day_key.txt", sep='\t')

genotype_info_wallace2018_df.set_index('Genotype', inplace=True)
genotype_info_wallace2018_df.head()

Plot_Day
Genotype             
33-16     X14A0309_26
38-11      X14A0233_8
4226      X14A0079_26
4722       X14A0311_8
A188       X14A0021_8

In [29]:
merged_df = genotype_info_wallace2018_df.merge(genotype_info_flintgarcia_df, left_index=True, right_index=True, how='left')
merged_df.head()

Plot_Day State/Country                    Pedigree    NSS     SS  \
Genotype                                                                        
33-16     X14A0309_26       Indiana   Lux Johnson Country white  0.972  0.014   
38-11      X14A0233_8       Indiana  Outcross in line from 176A  0.993  0.003   
4226      X14A0079_26      Illinois                 Funk 90 Day  0.917  0.071   
4722       X14A0311_8       Indiana              SA x Ohio yel.      0      0   
A188       X14A0021_8     Minnesota          [(4-29*64)4-29(4)]  0.982  0.013   

             TS Pop A Sweet A Subpopulation  
Genotype                                     
33-16     0.014     0       0           nss  
38-11     0.004     0       0           nss  
4226      0.012     0       0           nss  
4722          0     1       0       popcorn  
A188      0.006     0       0           nss

In [33]:
# Counting the number of genotypes for each group
merged_df['Subpopulation'].value_counts()

Subpopulation
nss        102
mixed       58
ts          46
ss          39
popcorn      9
sweet        5
Name: count, dtype: int64

* nss = Non stiff-stalk
* ts = Tropical/subtropical
* ss = Stiff-Stalk

In [34]:
merged_df['Subpopulation'].value_counts().sum()

259

In [31]:
merged_df.loc[merged_df['Subpopulation'].isna()]

Plot_Day State/Country Pedigree  NSS   SS   TS Pop A  \
Genotype                                                                      
CML206              X14A0564_26           NaN      NaN  NaN  NaN  NaN   NaN   
CML330              X14A0565_26           NaN      NaN  NaN  NaN  NaN   NaN   
CML411              X14A0566_26           NaN      NaN  NaN  NaN  NaN   NaN   
CML418               X14A0567_8           NaN      NaN  NaN  NaN  NaN   NaN   
CML504              X14A0569_26           NaN      NaN  NaN  NaN  NaN   NaN   
CML505               X14A0570_8           NaN      NaN  NaN  NaN  NaN   NaN   
CML511              X14A0571_26           NaN      NaN  NaN  NaN  NaN   NaN   
CML84               X14A0572_26           NaN      NaN  NaN  NaN  NaN   NaN   
CML85               X14A0573_26           NaN      NaN  NaN  NaN  NaN   NaN   
CML96               X14A0574_26           NaN      NaN  NaN  NaN  NaN   NaN   
Il101T               X14A0139_8           NaN      NaN  NaN  NaN  NaN   NaN   
IllHy               X14A0243_26           NaN      NaN  NaN  NaN  NaN   NaN   
MR19_Santo_Domingo   X14A0576_8           NaN      NaN  NaN  NaN  NaN   NaN   
MR20_Shoepeg         X14A0577_8           NaN      NaN  NaN  NaN  NaN   NaN   
W22R-rstd           X14A0240_26           NaN      NaN  NaN  NaN  NaN   NaN   
W22R-rstd            X14A0240_8           NaN      NaN  NaN  NaN  NaN   NaN   
Yu796NS              X14A0127_8           NaN      NaN  NaN  NaN  NaN   NaN   

                   Sweet A Subpopulation  
Genotype                                  
CML206                 NaN           NaN  
CML330                 NaN           NaN  
CML411                 NaN           NaN  
CML418                 NaN           NaN  
CML504                 NaN           NaN  
CML505                 NaN           NaN  
CML511                 NaN           NaN  
CML84                  NaN           NaN  
CML85                  NaN           NaN  
CML96                  NaN           NaN  
Il101T                 NaN           NaN  
IllHy                  NaN           NaN  
MR19_Santo_Domingo     NaN           NaN  
MR20_Shoepeg           NaN           NaN  
W22R-rstd              NaN           NaN  
W22R-rstd              NaN           NaN  
Yu796NS                NaN           NaN

In [32]:
merged_df['Subpopulation'].isna().sum()

17

In Romay paper, the following inbred are classified:
* CML206 (tropical)
* CML330 (tropical)
* CML418 (tropical)
* Il101T (sweet corn)
* IllHy (stiff stalk)
* MR19 Santo Domingo (landraces)
* MR20 Shoepeg (landraces)
* Yu796NS (unclassified)

# Summary

## Romay (2013)

* 118 genotypes have a group based on Romay et al (2013)
* 151 are unclassified (also based on Romay et al. 2013)
* Seven genotypes are not included in the table (or have a different identifier)

Initial correlation analyses considering maize groups will be using the 118 genotypes above (= classified in one of the subpopulations).

## Flint-Garcia (2005)

* 259 genotypes have a group based on Flint-Garcia (2005)
* A few genotypes (seven) were classified by Romay et al. (2013)
* For ten genotypes group information is missing